test-runner-poc/environments.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+const semver = require('semver');
+const path = require('path');
+const core = require('../lib/core');
+/**
+ * Base class for environments in projects that use the test
+ * runner
+ */
+class EnvironmentBase {
+  /**
+   * The default implementation of the environment setup
+   *
+   * @param {*} callback
+   */
+  setup(callback) {
+    callback();
+  }
+  constructor(parsedURI) {
+    this.host = parsedURI.hosts[0].host;
+    this.port = parsedURI.hosts[0].port;
+  }
+}
+const genReplsetConfig = (port, options) => {
+  return Object.assign(
+    {
+      options: {
+        bind_ip: 'localhost',
+        port: port,
+        dbpath: `${__dirname}/../db/${port}`,
+        setParameter: ['enableTestCommands=1']
+      }
+    },
+    options
+  );
+};
+function usingUnifiedTopology() {
+  return !!process.env.MONGODB_UNIFIED_TOPOLOGY;
+}
+/**
+ *
+ * @param {*} discoverResult
+ */
+class ReplicaSetEnvironment extends EnvironmentBase {
+  constructor(parsedURI, version) {
+    super(parsedURI);
+    this.setName = 'replset';
+    this.url = `mongodb://%s${this.host}:${this.port}/integration_tests?rs_name=rs`;
+    this.writeConcernMax = { w: 'majority', wtimeout: 30000 };
+    this.replicasetName = 'rs';
+    this.topology = function(topologyHost, topologyPort, options) {
+      topologyHost = topologyHost || this.host;
+      topologyPort = topologyPort || this.port;
+      options = Object.assign({}, options);
+      options.replicaSet = 'rs';
```

Comment:
What does this option do?

Reply:
I removed that line and ran the tests that used `options.replicaSet` but everything still ran smoothly so I deleted it!